docs/streams/upgrade-guide.html

```diff
@@ -141,6 +141,22 @@ <h3 class="anchor-heading"><a id="streams_notable_changes" class="anchor-link"><
 
     <h3><a id="streams_api_changes_400" href="#streams_api_changes_400">Streams API changes in 4.0.0</a></h3>
 
+    <p>
+        In this release, eos-v1 (Exactly Once Semantics version 1) is no longer supported. Additionally, all deprecated methods, classes, APIs, and config parameters
+        up to (including) AK 3.5 release have been removed.
```

Comment:
Is listing all the removed APIs necessary?

Reply:
Good question... It might be worth to highlight a few important ones like old Processor API, KStream#[flat]Transform[Values], KStream#through, KStream#branch,  Time/Session/SlidingWindows#of(), and KafkaStreams#setUncaughtExceptionHandler ?)

And it might be good to link to the Jira ticket for details: https://issues.apache.org/jira/browse/KAFKA-12822

For EOSv1, we should also call out that EOSv2 requires broker version 2.5+.